home/farcall.asm

```diff
@@ -2,131 +2,233 @@
 ; Functions that rely on "following" data access it via their return address on
 ; the stack, and update the return address to point past the data.
 
-MACRO update_pushed_af_flags
-; Must have been preceded by 'push af'.
-; Updates the flags on the stack to their current value
-; without clobbering any other registers.
-	push af ; pushes current flags
+_RstFarCall::
+	call .do_farcall
+	jr _ReturnFarCall
+; Call the following dab pointer
+; preserves af, bc, de, hl
+
+; Stack layout
```

Comment:
It's not clear at what point this layout is supposed to actually be the case during all the inc/dec sps and push/pops below.

Reply:
After the first set of dec sp+push, should i move the comment there?

Reply:
If that's the point where it's correct, yes please.